cpp/dolfinx/mesh/utils.h

```diff
@@ -888,6 +888,234 @@ Mesh<typename std::remove_reference_t<typename U::value_type>> create_mesh(
               std::move(geometry));
 }
 
+/// @brief Create a distributed mixed-topology mesh from mesh data using a
+/// provided graph partitioning function for determining the parallel
+/// distribution of the mesh.
+///
+/// From mesh input data that is distributed across processes, a
+/// distributed mesh::Mesh is created. If the partitioning function is
+/// not callable, i.e. it does not store a callable function, no
+/// re-distribution of cells is done.
+///
+/// @note This is an experimental specialised version of `create_mesh` for mixed
+/// topology meshes, and does not include cell reordering.
+///
+/// @param[in] comm Communicator to build the mesh on.
+/// @param[in] commt Communicator that the topology data (`cells`) is
+/// distributed on. This should be `MPI_COMM_NULL` for ranks that should
+/// not participate in computing the topology partitioning.
+/// @param[in] cells Cells on the calling process. Each cell is defined by its
```

Comment:
Needs improved docs - the type is now 'array of arrays'. Docs need to explain what's what.